dashboard/app/controllers/script_levels_controller.rb

```
@@ -175,6 +175,21 @@ def self.get_script_level(script, params)
     end
   end
 
+  # Get a JSON summary of a level's information, used in modern labs that don't
+  # reload the page between level views.  Note that this can be cached for a relatively
+  # long amount of time, including by the CDN, and does not vary per user.
+  def level_data
+    authorize! :read, ScriptLevel
+
+    @script = ScriptLevelsController.get_script(request)
+    @script_level = ScriptLevelsController.get_script_level(@script, params)
+    raise ActiveRecord::RecordNotFound unless @script_level
+
+    @level = @script_level.level
+
+    render json: {level_data: @level.properties["level_data"]}
```

Should we return an error if the level doesn't have any level_data? or let the client handle that?

---

Not sure yet, but I think the client might be able to handle it.  There might even be a case in which no `level_data` is valid, and just means the lab runs in some kind of default state...